src/build/_util.py

```diff
@@ -65,3 +67,18 @@ def check_dependency(
 
 def parse_wheel_filename(filename: str) -> re.Match[str] | None:
     return _WHEEL_FILENAME_REGEX.match(filename)
+
+
+if typing.TYPE_CHECKING:
+    TarFile = tarfile.TarFile
+
+else:
+    # Per https://peps.python.org/pep-0706/, the "data" filter will become
+    # the default in Python 3.14.
+    if sys.version_info < (3, 14) and hasattr(tarfile, 'data_filter'):
+
+        class TarFile(tarfile.TarFile):
+            extraction_filter = tarfile.data_filter
```

Comment:
https://peps.python.org/pep-0706/

> Applications and system integrators may wish to change extraction_filter of the TarFile class itself to set a global default. When using a function, they will generally want to wrap it in staticmethod() to prevent injection of a self argument.

```

Traceback (most recent call last):
  File "/builddir/build/BUILDROOT/python-build-0.10.0-5.fc39.x86_64/usr/lib/python3.12/site-packages/build/__main__.py", line 376, in main
    built = build_call(
            ^^^^^^^^^^^
  File "/builddir/build/BUILDROOT/python-build-0.10.0-5.fc39.x86_64/usr/lib/python3.12/site-packages/build/__main__.py", line 243, in build_package_via_sdist
    t.extractall(sdist_out)
  File "/usr/lib64/python3.12/tarfile.py", line 2253, in extractall
    tarinfo = self._get_extract_tarinfo(member, filter_function, path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/tarfile.py", line 2305, in _get_extract_tarinfo
    tarinfo = filter_function(tarinfo, path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: data_filter() takes 2 positional arguments but 3 were given
```

Reply:
Thanks, fixed.

Reply:
Thanks for testing it!